php/example_code/aws_utilities/AWSServiceClass.php

```diff
@@ -43,4 +44,68 @@ public function customWaiter($function, $verbose = false)
         }
         return $result;
     }
+
+    /**
+     * This is a very common task for all services, so this is a helper function to use everywhere.
+     */
+
+    /**
+     * @param $roleName string the name to give the role
+     * @param string|null $rolePolicyDocument
+     * @param $clientArgs array|null optional array of client arguments
+     * @param $verbose bool whether to echo debug information
+     * @return string the created role's ARN
+     */
+    public function createRole(
+        string $roleName,
+        string $rolePolicyDocument = null,
+        array $clientArgs = null,
+        bool $verbose = false
+    ): string {
+        $clientArgs = $clientArgs ?: [
+            'region' => 'us-west-2',
+            'version' => 'latest',
+            'profile' => 'default',
+        ];
+        $iamService = new IAMService($clientArgs);
+
+        $rolePolicyDocument = $rolePolicyDocument ?: "{
+            \"Version\": \"2012-10-17\",
+            \"Statement\": [
+                {
+                    \"Effect\": \"Allow\",
+                    \"Principal\": {
+                        \"Service\": \"states.amazonaws.com\"
+                    },
+                    \"Action\": \"sts:AssumeRole\"
+                }
+            ]
+        }";
+
+        $role = $iamService->createRole($roleName, $rolePolicyDocument);
+        echo $verbose ? "Created role {$role['RoleName']}.\n" : "";
+        $iamService->attachRolePolicy(
+            $role['RoleName'],
+            "arn:aws:iam::aws:policy/AWSStepFunctionsFullAccess"
+        );
+        echo $verbose ? "Attached the AWSStepFunctionsFullAccess to {$role['RoleName']}.\n" : "";
```

Comment:
Same comment about maybe parameterizing these parts so this function is more general purpose.

Reply:
Yes...but later